File: generate-sitemap.js
// generate-sitemap.js
const fs = require('fs');
const path = require('path');

// List of all the HTML files in your project (update this if your pages are in different directories)
const pages = [
  'index.html',
  'stadiumstay.html',
  'about.html',
  'blog.html',
  'calculator.html',
  'contact.html',
  'faq.html',
  'privacy.html',
  'terms.html',
  'destinations.html',
  'collections.html',
  'routes.html',
  'solo.html',
  'surge.html',
  'subscribe.html',
  // Add more HTML pages as needed
];

// Function to generate the sitemap
function generateSitemap() {
  const urlSet = pages.map(page => {
    const lastModified = new Date().toISOString(); // Set this dynamically if you have last-modified information
    return `
      <url>
        <loc>https://brightlane.github.io/SkyScanner/${page}</loc>
        <lastmod>${lastModified}</lastmod>
        <changefreq>daily</changefreq>
        <priority>0.8</priority>
      </url>
    `;
  }).join('\n');

  const sitemap = `<?xml version="1.0" encoding="UTF-8"?>
  <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
    ${urlSet}
  </urlset>`;

  // Save the sitemap.xml to the root directory
  fs.writeFileSync(path.join(__dirname, 'sitemap.xml'), sitemap);
  console.log('Sitemap generated successfully!');
}

// Execute the function
generateSitemap();

File: scripts/generate_sitemap.py
import os
from datetime import datetime

# Configuration - Update this to your actual GitHub Pages URL
BASE_URL = "https://brightlane.github.io/SkyScanner" 
FLIGHTS_DIR = 'flights'
SITEMAP_FILE = 'sitemap.xml'

def build_sitemap():
    pages = []
    
    # 1. Add the main index if it exists
    pages.append(BASE_URL)

    # 2. Scan the flights directory for all generated route pages
    if os.path.exists(FLIGHTS_DIR):
        for file in os.listdir(FLIGHTS_DIR):
            if file.endswith(".html"):
                # Clean up the path for the URL
                pages.append(f"{BASE_URL}/{FLIGHTS_DIR}/{file}")

    # 3. Generate XML structure
    xml = '<?xml version="1.0" encoding="UTF-8"?>\n'
    xml += '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">\n'
    
    for page in pages:
        xml += '  <url>\n'
        xml += f'    <loc>{page}</loc>\n'
        xml += f'    <lastmod>{datetime.now().strftime("%Y-%m-%d")}</lastmod>\n'
        xml += '    <priority>0.8</priority>\n'
        xml += '  </url>\n'
    
    xml += '</urlset>'

    # 4. Write the sitemap to the root directory
    with open(SITEMAP_FILE, 'w', encoding='utf-8') as f:
        f.write(xml)
    
    print(f"🗺️ Vulture 10K: Sitemap updated with {len(pages)} URLs.")

if __name__ == "__main__":
    build_sitemap()
